float32 x
float32 y
float32 vel_x
float32 vel_y
float32 goal_x
float32 goal_y